internal/controller/storage/storage.go

```diff
@@ -0,0 +1,95 @@
+package storage
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+
+	danaiov1alpha1 "github.com/dana-team/nfspvc-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	nfsPvcDanaLabel = "nfspvc.dana.io/nfspvc-owner"
+)
+
+func PreparePVC(nfspvc danaiov1alpha1.NfsPvc, StorageClass string) corev1.PersistentVolumeClaim {
+	storageClass := StorageClass
+	return corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nfspvc.Name,
+			Namespace: nfspvc.Namespace,
+			Labels: map[string]string{
+				nfsPvcDanaLabel: nfspvc.Name,
+			},
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: &storageClass,
+			VolumeName:       nfspvc.Name + "-" + nfspvc.Namespace + "-pv",
+			AccessModes:      nfspvc.Spec.AccessModes,
+			Resources: corev1.VolumeResourceRequirements{
+				Requests: nfspvc.Spec.Capacity,
+			},
+		},
+	}
+}
+
+func PreparePV(nfspvc danaiov1alpha1.NfsPvc, StorageClass string, ReclaimPolicy string) corev1.PersistentVolume {
```

Comment:
Missing comment

Reply:
It doesn't actually create a new PV, it just returns a PersistentVolume object as per given parameters